Documentation/tls/tls-kube.md

```diff
@@ -0,0 +1,80 @@
+# Enabling custom Kubernetes TLS certificates
+
+Use Tectonic's [Kubernetes TLS module][kube-module] to enable user provided Kubernetes certificates.
+
+The module does not contain any logic, but just passes user provided certificates from its input directly to its output. This is to prevent changing existing references to the `tls/kube/self-signed` module, hence all `tls/kube/*` modules share the same outputs.
+
+## Usage
+
+Comment out the existing self-signed kube TLS in your platform, i.e. `platforms/aws/tectonic.tf`:
+
+```
+/*
+module "kube_certs" {
+  source = "../../modules/tls/kube/self-signed"
+
+  ca_cert_pem        = "${var.tectonic_ca_cert}"
+  ca_key_alg         = "${var.tectonic_ca_key_alg}"
+  ca_key_pem         = "${var.tectonic_ca_key}"
+  kube_apiserver_url = "https://${module.masters.api_internal_fqdn}:443"
+  service_cidr       = "${var.tectonic_service_cidr}"
+}
+*/
+```
+
+Configure the user provided certificate paths in your platform, i.e. `platforms/aws/tectonic.tf`:
+
+```
+module "kube_certs" {
+  source = "../../modules/tls/kube/user-provided"
+
+  ca_cert_pem_path        = "/path/to/ca.crt"
+  kubelet_cert_pem_path   = "/path/to/kubelet.crt"
+  kubelet_key_pem_path    = "/path/to/kubelet.key"
+  apiserver_cert_pem_path = "/path/to/apiserver.crt"
+  apiserver_key_pem_path  = "/path/to/apiserver.key"
+}
+```
+
+The signed kubelet certificate must have the following key usage associations:
```

Comment:
note. That all kubelets will use this cert if enabled. Otherwise each node will generate it's own certificate.